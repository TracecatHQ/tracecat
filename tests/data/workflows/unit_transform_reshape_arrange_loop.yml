title: Reshape data in a loop
description: Test reshaping data from a list of mappings to a list of key-value pairs
entrypoint:
  ref: flatten
inputs:
  contacts:
    - name: John Doe
      age: 30
      job:
        title: Software Engineer
        company: Acme Inc.
        city: Sometown
      residence:
        street_address: 123 Main St.
        city: Anytown
      contact:
        email: john@acme.com
        phone: 123-456-7890
    - name: Jane Doe
      age: 25
      job:
        title: Data Scientist
        company: Beta Corp.
        city: Othertown
      residence:
        street_address: 456 Elm St.
        city: Anothertown
      contact:
        email: janed@datacorp.com
        phone: 987-654-3210
    - name: Jim Doe
      age: 35
      job:
        title: DevOps Engineer
        company: Gamma LLC
        city: Yetanothertown
      residence:
        street_address: 789 Oak St.
        city: Yetanothercity
      contact:
        email: jim@gamma.com
        phone: 456-789-0123

actions:
  - ref: flatten
    action: core.transform.reshape
    for_each: ${{ for var.person in INPUTS.contacts }}
    # Flatten the mapping into a list of key-value pairs
    args:
      value:
        name: ${{ var.person.name }}
        age: ${{ var.person.age }}
        job_title: ${{ var.person.job.title }}
        job_company: ${{ var.person.job.company }}
        job_city: ${{ var.person.job.city }}
        residence_street_address: ${{ var.person.residence.street_address }}
        residence_city: ${{ var.person.residence.city }}
        contact_email: ${{ var.person.contact.email }}
        contact_phone: ${{ var.person.contact.phone }}
