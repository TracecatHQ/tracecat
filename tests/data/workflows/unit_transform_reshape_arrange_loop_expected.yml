ACTIONS:
  flatten:
    result:
      - name: John Doe
        age: 30
        job_title: Software Engineer
        job_company: Acme Inc.
        job_city: Sometown
        residence_street_address: 123 Main St.
        residence_city: Anytown
        contact_email: john@acme.com
        contact_phone: 123-456-7890
      - name: Jane Doe
        age: 25
        job_title: Data Scientist
        job_company: Beta Corp.
        job_city: Othertown
        residence_street_address: 456 Elm St.
        residence_city: Anothertown
        contact_email: janed@datacorp.com
        contact_phone: 987-654-3210
      - name: Jim Doe
        age: 35
        job_title: DevOps Engineer
        job_company: Gamma LLC
        job_city: Yetanothertown
        residence_street_address: 789 Oak St.
        residence_city: Yetanothercity
        contact_email: jim@gamma.com
        contact_phone: 456-789-0123
    result_typename: "list"

INPUTS:
  contacts:
    - name: John Doe
      age: 30
      job:
        title: Software Engineer
        company: Acme Inc.
        city: Sometown
      residence:
        street_address: 123 Main St.
        city: Anytown
      contact:
        email: john@acme.com
        phone: 123-456-7890
    - name: Jane Doe
      age: 25
      job:
        title: Data Scientist
        company: Beta Corp.
        city: Othertown
      residence:
        street_address: 456 Elm St.
        city: Anothertown
      contact:
        email: janed@datacorp.com
        phone: 987-654-3210
    - name: Jim Doe
      age: 35
      job:
        title: DevOps Engineer
        company: Gamma LLC
        city: Yetanothertown
      residence:
        street_address: 789 Oak St.
        city: Yetanothercity
      contact:
        email: jim@gamma.com
        phone: 456-789-0123
TRIGGER:
